src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionRequestHandler.cs

```diff
@@ -139,13 +152,20 @@ protected DataCollectionRequestHandler(IMessageSink messageSink)
                 {
                     if (Instance == null)
                     {
+                        var requestData = new RequestData
+                        {
+                            MetricsCollection = new NoOpMetricsCollection(),
+                            IsTelemetryOptedIn = false
```

Comment:
Does this belong in the constructor of RequestData instead? Seems odd adding it in here.

Reply:
Added default constructor